tests/Tests/Properties/Text.hs

```diff
@@ -431,6 +417,16 @@ testText =
         testProperty "ascii_toCaseFold" ascii_toCaseFold
       ],
 
+#if MIN_VERSION_base(4, 15, 0)
+      -- Requires matching version of Unicode in base and text
+      testGroup "char case conversion" $ if T.unicodeVersion == G.unicodeVersion then [
```

Comment:
FWIW, it's very easy to get into state where those tests are not run at all. Any upgrade's to Unicode version used by `text` will cause these tests to not be run, as GHC is (usually) behind.

And you kind of want these tests to be run when Unicode stuff is changed...